components/eamxx/src/share/io/scream_output_manager.cpp

```diff
@@ -636,16 +637,16 @@ compute_filename (const IOFileSpecs& file_specs,
   auto ts = (m_avg_type==OutputAvgType::Instant || file_specs.ftype==FileType::HistoryRestart)
           ? timestamp : control.last_write_ts;
 
+  int ts_string_len = 0;
   switch (file_specs.storage.type) {
-    case NumSnaps:
-      filename += "." + ts.to_string(); break;
-    case Yearly:
-      filename += "." + std::to_string(ts.get_year()); break;
-    case Monthly:
-      filename += "." + std::to_string(ts.get_year()) + "-" + std::to_string(ts.get_month()); break;
```

Comment:
This was also a bug: `std::to_string(1)` does not give `01`, but `1`, so we would get strings whose format is not exactly the same for, say, Jan and Dec. The method `TimeStamp::to_string` takes care of padding with 0's, so we're better off simply truncating that depending on our needs.